client/blocks/plan-thank-you-card/index.jsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
+import Gridicon from 'components/gridicon';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { fetchSitePlans } from 'state/sites/plans/actions';
+import { getPlanBySlug, getPlans } from 'state/plans/selectors';
+import { requestPlans } from 'state/plans/actions';
+
+class PlanThankYouCard extends Component {
+	static propTypes = {
+		selectedSite: PropTypes.object
+	};
+
+	componentDidMount() {
+		if ( ! this.props.plan ) {
+			this.props.fetchSitePlans( this.props.selectedSite );
+		}
+
+		if ( ! this.props.plansFetched ) {
+			this.props.requestPlans();
+		}
+	}
+
+	render() {
+		// Non standard gridicon sizes are used here because we use them as background pattern with various sizes and rotation
+		/* eslint-disable wpcalypso/jsx-gridicon-size */
+		return (
+			<div className="plan-thank-you-card">
+				<div className="plan-thank-you-card__header">
+					<Gridicon className="plan-thank-you-card__main-icon" icon="checkmark-circle" size={ 140 } />
+					<div className="plan-thank-you-card__plan-name">
+
+					</div>
+					{ ( ! this.props.planDetails ) ? (
+						<div className="plan-thank-you-card__plan-name is-placeholder"></div>
+					) : (
+						<div className="plan-thank-you-card__plan-name">
+							{ this.props.translate( '%(planName)s Plan', {
+								args: { planName: this.props.planDetails.product_name_short }
```

Comment:
Please translate the plan name, this will be incorrect in other languages.

Also see if existing plan constants would be useful to you: https://github.com/Automattic/wp-calypso/blob/master/client/lib/plans/constants.js#L124